### PR TITLE
[border-agent] add logging for border agent events

### PR DESCRIPTION
**Pull request by wgtdkp**
This PR includes several small changes:
1. Add logs for Border Agent `Start` and `Stop` events to help analysis commissioning issues such as https://github.com/openthread/ot-commissioner/issues/193#issuecomment-822120350.
2. Start Border Agent when the native commissioner is stopped.
3. Add CLI command `ba state` to get current Border Agent state.

### CODE REVIEW

**Review by bukepo — COMMENTED** [review]
I guess this will conflict with https://github.com/openthread/openthread/pull/6450. Can we hold up this PR until #6450 is merged?

**Comment by wgtdkp** [comment]
> I guess this will conflict with #6450. Can we hold up this PR until #6450 is merged?

OK to me, not an urgent PR.

**Review by simonlingoogle — APPROVED** [review]
LGTM :+1:

**Review by bukepo — APPROVED** [review]

**Review by jwhui — APPROVED** [review]

**Pull request merged**
